File: mobile/src/main/java/com/novoda/materialised/hackernews/asynclistview/ModelledViewHolder.java
package com.novoda.materialised.hackernews.asynclistview;

import android.support.v7.widget.RecyclerView;
import android.view.View;

final class ModelledViewHolder<V extends View & ModelledView> extends RecyclerView.ViewHolder {

    private final V heldModelledView;

    ModelledViewHolder(V itemView) {
        super(itemView);
        heldModelledView = itemView;
    }

    V obtainHeldView() {
        return heldModelledView;
    }
}

File: mobile/src/main/java/com/novoda/materialised/hackernews/asynclistview/ModelledViewHolder.kt
package com.novoda.materialised.hackernews.asynclistview

import android.view.View
import androidx.recyclerview.widget.RecyclerView

internal class ModelledViewHolder<V>(private val heldModelledView: V) : RecyclerView.ViewHolder(heldModelledView)
    where V : View, V : ModelledView<*> {

    fun obtainHeldView(): V {
        return heldModelledView
    }
}
